lib/mongoid/includes/inclusion.rb

```diff
@@ -17,6 +17,14 @@ def nested?
         !!from
       end
 
+      # Public: Checks if the collection already has an inclusion with the
+      # specified metadata.
+      def eql?(other)
+        metadata == other && from == other.from
+      rescue
+        false
```

Comment:
When comparing an inclusion with `other`, `other` can be anything so the `from` method is not always available (will raise a `NoMethodError` in that case).

Normally you could prevent this using `other.try(:from)` but that doesn't work well with the delegator structure, as the `try` method will be delegated to the metadata object which doesn't have the `from` method.

In this case you would want the `try` method to be forced to not be delegated, but I couldn't find a way to do that so rescueing the exception is the only solution I could think of.